### PR TITLE
Rename Shared CPU Mode to Shared Core Mode

### PR DESCRIPTION
**Pull request by koolzz**
Current name doesn't fit well as NFs are sharing cores.

## Summary:
Ran cscope and changed everything, hopefully didn't miss anything

| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                |
| New functionality        |
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 

## Merging notes:
 - Dependencies: None  

**TODO before merging :**
 - [x] PR is ready for review

## Review: 
(optional) << @-mention people who should review these changes >>

(optional) **Subscribers:** << @-mention people who probably care about these changes >>

### CODE REVIEW

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Your results will arrive shortly

**Comment by onvm** [comment]
> In response to PR creation

## CI Message

Error: ERROR: Failed to fetch results from nimbnode30


## Linter Output

onvm/onvm_nflib/onvm_common.h:70:  Lines should be <= 120 characters long  [whitespace/line_length] [5]
Total errors found: 1

File: onvm/onvm_nflib/onvm_common.h [hunk 1]
```diff
@@ -67,14 +67,14 @@
 
 #define PACKET_READ_SIZE ((uint16_t)32)
 
-#define ONVM_ENABLE_SHARED_CPU_DEFAULT 0  // default value for shared cpu logic, if true NFs sleep while waiting for packets
+#define ONVM_NF_CORE_SHARING_DEFAULT 0  // default value for shared core logic, if true NFs sleep while waiting for packets
```
Comment (kevindweb) [inline onvm/onvm_nflib/onvm_common.h]: Can we change the comment to `// default for shared core logic, if true NFs sleep while waiting for packets`. Linter would stop complaining, right now we're at 123 characters

Comment (koolzz) [reply onvm/onvm_nflib/onvm_common.h]: Yeah good point, will fix

File: onvm/onvm_mgr/onvm_stats.c [hunk 2]
```diff
@@ -566,7 +566,7 @@ onvm_stats_display_nfs(unsigned difftime, uint8_t verbosity_level) {
                 }
         }
 
-        if (ONVM_ENABLE_SHARED_CPU) {
+        if (ONVM_NF_CORE_SHARING) {
```
Comment (kevindweb) [inline onvm/onvm_mgr/onvm_stats.c]: Can we change the next line to `fprintf(stats_out, "\n\nShared Core stats\n");` while we're here?

**Comment by koolzz** [comment]
@onvm go buddy

**Comment by onvm** [comment]
> @onvm go buddy

## CI Message

Your results will arrive shortly

**Pull request merged**